InnerEye/ML/SSL/lightning_modules/ssl_classifier_module.py

```diff
@@ -0,0 +1,89 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+from typing import Any, List, Optional
+
+import torch
+from pl_bolts.models.self_supervised import SSLEvaluator
+from torch.nn import functional as F
+
+from InnerEye.ML.SSL.encoders import get_encoder_output_dim
+from InnerEye.ML.dataset.scalar_sample import ScalarItem
+from InnerEye.ML.lightning_container import LightningModuleWithOptimizer
+from InnerEye.ML.lightning_metrics import Accuracy05, AreaUnderPrecisionRecallCurve, AreaUnderRocCurve, \
+    ScalarMetricsBase
+from InnerEye.ML.utils.device_aware_module import DeviceAwareModule
+
+
+class SSLClassifier(LightningModuleWithOptimizer, DeviceAwareModule):
+    """
+    SSL Image classifier that combines pre-trained SSL encoder with a trainable linear-head.
+    """
+
+    def __init__(self,
+                 num_classes: int,
+                 encoder: torch.nn.Module,
+                 freeze_encoder: bool,
+                 class_weights: Optional[torch.Tensor]):
+        super().__init__()
+        self.num_classes = num_classes
+        self.encoder = encoder
+        self.freeze_encoder = freeze_encoder
+        self.class_weights = class_weights
+        self.encoder.eval()
+        self.classifier_head = SSLEvaluator(n_input=get_encoder_output_dim(self.encoder),
+                                            n_hidden=None,
+                                            n_classes=num_classes,
+                                            p=0.20)
+        self.train_metrics: List[ScalarMetricsBase] = [AreaUnderRocCurve(), AreaUnderPrecisionRecallCurve(),
```

Comment:
Do you mind if we format it differently for the ease of reading?
`self.train_metrics: List[ScalarMetricsBase] = \
   [AreaUnderRocCurve(), AreaUnderPrecisionRecallCurve(), Accuracy05()] if self.num_classes == 2 else [Accuracy05()]

Reply:
or something similar if possible

Reply:
separated completely class 2 and class > 2 to make it clearer